firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteLocalDocumentCache.java

```diff
@@ -0,0 +1,210 @@
+package com.google.firebase.firestore.local;
+
+import static com.google.firebase.firestore.util.Assert.fail;
+import static com.google.firebase.firestore.util.Assert.hardAssert;
+
+import com.google.firebase.database.collection.ImmutableSortedMap;
+import com.google.firebase.firestore.auth.User;
+import com.google.firebase.firestore.core.Query;
+import com.google.firebase.firestore.model.DocumentCollections;
+import com.google.firebase.firestore.model.DocumentKey;
+import com.google.firebase.firestore.model.MutableDocument;
+import com.google.firebase.firestore.model.ResourcePath;
+import com.google.firebase.firestore.util.BackgroundQueue;
+import com.google.firebase.firestore.util.Executors;
+import com.google.protobuf.InvalidProtocolBufferException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+public class SQLiteLocalDocumentCache implements LocalDocumentCache {
+  private final SQLitePersistence db;
+  private final String uid;
+  private final LocalSerializer serializer;
+
+  SQLiteLocalDocumentCache(SQLitePersistence persistence, User user, LocalSerializer serializer) {
+    this.db = persistence;
+    this.uid = user.isAuthenticated() ? user.getUid() : "";
+    this.serializer = serializer;
+  }
+
+  @Override
+  public void add(MutableDocument doc) {
+    if (doc == null) return;
+    String path = EncodedPath.encode(doc.getKey().getPath());
+    int mutationFlag = 0;
+    if (doc.hasLocalMutations()) {
+      mutationFlag |= 1;
+    }
+    if (doc.hasCommittedMutations()) {
+      mutationFlag |= 2;
+    }
```

Comment:
We can just store these values in two columns or encode them in the MaybeDocument proto.

Reply:
I think encoding them in MaybeDocument proto might make more sense.